icu4c/source/common/uloc.cpp

```diff
@@ -906,34 +939,33 @@ uloc_setKeywordValue(const char* keywordName,
     }
     keywordValueBuffer[keywordValueLen] = 0; /* terminate */
 
-    startSearchHere = (char*)locale_getKeywordsStart(buffer);
-    if(startSearchHere == nullptr || (startSearchHere[1]==0)) {
+    if (keywords == nullptr || keywords[1] == '\0') {
         if(keywordValueLen == 0) { /* no keywords = nothing to remove */
             U_ASSERT(*status != U_STRING_NOT_TERMINATED_WARNING);
-            return bufLen;
+            return 0;
```

Comment:
Should this return 1 if keywords != nullptr because there is something (a separator character?) in keywords[0]?

Reply:
I think it should be the way it is, the zero return value here is the status code for _do nothing_.

Reply:
Is that documented somewhere? When this code was part of the API function, it used to return bufLen for the length of the result. Where does "less-or-equal-zero is a status code" come from?

Reply:
I don't think that's documented anywhere, it's just preserving existing behaviour (cf. my other comment with the code snippet).